main.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/bitrise-io/go-steputils/stepconf"
+	"github.com/bitrise-io/go-steputils/tools"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/env"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/retry"
+	"github.com/bitrise-io/go-xcode/autocodesign"
+	"github.com/bitrise-io/go-xcode/autocodesign/certdownloader"
+	"github.com/bitrise-io/go-xcode/autocodesign/codesignasset"
+	"github.com/bitrise-io/go-xcode/autocodesign/devportalclient"
+	"github.com/bitrise-io/go-xcode/autocodesign/keychain"
+	"github.com/bitrise-io/go-xcode/autocodesign/projectmanager"
+	"github.com/bitrise-io/go-xcode/devportalservice"
+)
+
+func failf(format string, args ...interface{}) {
+	log.Errorf(format, args...)
+	os.Exit(1)
+}
+
+func main() {
+	// Parse and validate inputs
+	var cfg Config
+	parser := stepconf.NewInputParser(env.NewRepository())
+	if err := parser.Parse(&cfg); err != nil {
+		failf("Config: %s", err)
+	}
+	stepconf.Print(cfg)
+
+	log.SetEnableDebugLog(cfg.VerboseLog)
+
+	certsWithPrivateKey, err := cfg.Certificates()
+	if err != nil {
+		failf("Failed to convert certificate URLs: %s", err)
+	}
+
+	// Analyze project
+	fmt.Println()
+	log.Infof("Analyzing project")
+	project, err := projectmanager.NewProject(projectmanager.InitParams{
+		ProjectOrWorkspacePath: cfg.ProjectPath,
+		SchemeName:             cfg.Scheme,
+		ConfigurationName:      cfg.Configuration,
+	})
+	if err != nil {
+		failf(err.Error())
+	}
+
+	appLayout, err := project.GetAppLayout(cfg.SignUITestTargets)
+	if err != nil {
+		failf(err.Error())
+	}
+
+	// Create Apple developer Portal client
+	clientType, err := parseClientType(cfg.BitriseConnection)
+	if err != nil {
+		failf("Invalid input: unexpected value for Bitrise Apple Developer Connection (%s)", cfg.BitriseConnection)
+	}
+
+	var connection devportalservice.AppleDeveloperConnection
+	isRunningOnBitrise := cfg.BuildURL != "" && cfg.BuildAPIToken != ""
+
+	switch {
+	case !isRunningOnBitrise:
+		fmt.Println()
+		log.Warnf("Connected Apple Developer Portal Account not found. Step is not running on bitrise.io: BITRISE_BUILD_URL and BITRISE_BUILD_API_TOKEN envs are not set")
```

Comment:
Should probably fail in this case. Maybe even mention the `file://` URL schema for the BITRISE_BUILD_URL Input.

Reply:
Yeah but you also have to provide a "fake" api token?